src/lib.rs

```diff
@@ -9,30 +9,25 @@
 // except according to those terms.
 
 #![no_std]
-#![cfg_attr(feature = "alloc", feature(alloc))]
+#![cfg_attr(feature = "nightly", feature(cfg_target_has_atomic))]
 
 #[cfg(test)]
 #[macro_use]
 extern crate std;
 
-#[cfg(feature = "alloc")]
 extern crate alloc;
 extern crate cty;
 extern crate memchr;
 
-#[cfg(feature = "alloc")]
+#[cfg(feature = "nightly")]
+#[cfg(target_has_atomic = "ptr")]
```

Comment:
I don't think this condition does what you describe. When the nightly feature is disabled, we want to enable `Arc` by default.

Try something along the lines of:
`#[cfg_attr(feature = "nightly", cfg(target_has_atomic = "ptr")]`

Reply:
This is tough. I was trying to come up with a solution that would allow both:
- building with stable and inclusion of Arc without the nightly-specific atomic check
- building with nightly with the 'nightly' feature to check atomic checks

The `cfg` setup is a little elaborate, and still exposes the failure case the `nightly` feature is specified when using the stable toolchain. But maybe that is OK.